broadcast/broadcast-client/client_builder.go

```diff
@@ -0,0 +1,30 @@
+package broadcast_client
+
+import (
+	broadcast_api "github.com/bitcoin-sv/go-broadcast-client/broadcast"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/internal/arc"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/internal/composite"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/internal/httpclient"
+)
+
+type builder struct {
+	factories []composite.BroadcastFactory
+}
+
+func Builder() *builder {
+	return &builder{}
+}
+
+func (cb *builder) WithArc(config ArcClientConfig, client httpclient.HTTPInterface) *builder {
```

Comment:
I'm not sure that `httpClient` should be passed here. 
Maybe you could change it and add `WithHttpClient()` function?

Reply:
done